packages/cli-kit/src/public/node/ruby.ts

```diff
@@ -293,13 +306,17 @@ async function bundleInstallThemeCheck() {
 /**
  * It returns the directory where the Ruby CLI is located.
  *
+ * @param embedded - True when embebbed codebase of CLI should be used.
  * @returns The absolute path to the directory.
  */
-function shopifyCLIDirectory(): string {
-  return (
-    process.env.SHOPIFY_CLI_2_0_DIRECTORY ??
-    joinPath(pathConstants.directories.cache.vendor.path(), 'ruby-cli', RubyCLIVersion)
-  )
+async function shopifyCLIDirectory(embedded = false): Promise<string> {
+  return embedded
+    ? ((await file.findPathUp('assets/cli-sources', {
```

Comment:
> Why do you need to find the path up if it's a fixed one?

We can do it using a relative path (see snippet below)  but I think is less readable a prone to errors, for example in case you want to move the package in a nested level.
```ts
  const embeddedDirectory = joinPath(dirname(fileURLToPath(import.meta.url)), '..', '..', '..', 'assets', 'cli-sources')
```


> Also, nitpicky suggestion: I always find the ternary operator hard to read when it's too long. What about extracting the path generation to other functions?

👍

Reply:
Couldn't we do something like `joinPath(moduleDirectory(import.meta.url), 'assets', 'cli-sources')`? We could even add `import.meta.url` as the default parameter in moduleDirectory to simply do `joinPath(moduleDirectory(), 'assets', 'cli-sources')`.